precompiles/ArbOwner.go

```diff
@@ -157,3 +161,50 @@ func (con ArbOwner) ReleaseL1PricerSurplusFunds(c ctx, evm mech, maxWeiToRelease
 	}
 	return weiToTransfer, nil
 }
+
+func (con ArbOwner) SetChainConfig(c ctx, evm mech, serializedChainConfig []byte) error {
+	if c == nil {
+		return errors.New("nil context")
+	}
+	if c.txProcessor == nil {
+		return errors.New("uninitialized tx processor")
+	}
+	if c.txProcessor.MsgIsGasEstimation() {
+		var newConfig params.ChainConfig
+		err := json.Unmarshal(serializedChainConfig, &newConfig)
+		if err != nil {
+			return fmt.Errorf("invalid chain config, can't deserialize: %w", err)
+		}
+		if newConfig.ChainID == nil {
+			return errors.New("invalid chain config, missing chain id")
+		}
+		// TODO do we want to validate chain id? Is a change of chain id a valid operation?
+		chainId, err := c.State.ChainId()
+		if err != nil {
+			return fmt.Errorf("failed to get chain id from ArbOS state: %w", err)
+		}
+		if newConfig.ChainID.Cmp(chainId) != 0 {
```

Comment:
I think a change of chain id is a theoretically possible operation, but the node doesn't support it, so I'd keep this check in-place. It helps prevent accidentally setting another chain's config on a chain. If in the future the node does support changing chain ids, then we can remove this check.